dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/api/RequestApi.kt

```diff
@@ -91,5 +95,75 @@ interface RequestApi {
     fun getAggregatedDataRequests(
         @RequestParam identifierValue: String? = null,
         @RequestParam dataTypes: Set<DataTypeEnum>? = null,
-    ): List<AggregatedDataRequest>
+        @RequestParam reportingPeriod: String? = null,
+    ): ResponseEntity<List<AggregatedDataRequest>>
+
+    /**
+     * A method to post a single request to Dataland.
+     * @param singleDataRequest includes necessary info for the single request
+     * @return response after posting a single data request to Dataland
+     *
+     */
+    @Operation(
+        summary = "Send a single request",
+        description = "A single of data requests for specific frameworks and companies is being sent.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully processed a single data request."),
+        ],
+    )
+    @PostMapping(
+        value = ["/single"],
+        consumes = ["application/json"],
+        produces = ["application/json"],
+    )
+    @PreAuthorize("hasRole('ROLE_USER')")
+    fun postSingleDataRequest(
+        @Valid @RequestBody
+        singleDataRequest: SingleDataRequest,
+    ):
+        ResponseEntity<List<StoredDataRequest>>
+
+    /** A method for users to get a data request by its ID.
+     * @return the data requests corresponding to the provided ID
+     */
+    @Operation(
+        summary = "Get all stored data requests of the user making the request.",
+        description = "Gets all the stored data request created by the user who is making the request.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully retrieved data request for the provided ID."),
+        ],
+    )
+    @GetMapping(
+        value = ["/{dataRequestId}"],
+        produces = ["application/json"],
+    )
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    fun getDataRequestById(@PathVariable dataRequestId: UUID): ResponseEntity<StoredDataRequest>
+
+    /** Changes request status of existing data request
+     * @return the modified data request
+     */
+
+    @Operation(
+        summary = "Update status of data request.",
+        description = "Updates status of data request given data request id.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully patched data request."),
+        ],
+    )
+    @PatchMapping(
+        value = ["/{dataRequestId}/requestStatus"],
+        produces = ["application/json"],
+    )
+    @PreAuthorize("hasRole('ROLE_USER')")
+    fun patchDataRequest(
```

Comment:
Is this intended? Feels somehow wrong that a normal user can patch the status of any data request.